### PR TITLE
Remove excess features

### LINKED ISSUE #1275: Remove excess dependency features from 2.0
## Description

There are many features in the SDK that are created by dependencies that should be removed, as they are polluting the feature space. They should be removed at 2.0 since this is a breaking change.
- `url`
- `instant`
- `iota-ledger-nano`
- `serde_repr`
- `regex`
- `rumqttc`
- `rand` (the feature is just a wrapper)
- `bs58`
- `iota_stronghold`
- `thiserror`
- `futures`
- `tokio`
- `zeroize`
- `reqwest`
- `async-trait`
- `log`

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Remove features that need not exist

## Links to any relevant issues

Closes #1275

### CODE REVIEW

File: sdk/src/utils/mod.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #[cfg(feature = "serde")]
 pub mod serde;
 
-#[cfg(feature = "instant")]
+#[cfg(feature = "client")]
 pub fn unix_timestamp_now() -> core::time::Duration {
```
Comment (thibault-martinez) [inline sdk/src/utils/mod.rs]: Move it to client/utils.rs ?

File: sdk/Cargo.toml [hunk 2]
```diff
@@ -138,20 +138,19 @@ tokio = { version = "1.32.0", default-features = false, features = [
...
-rand = ["dep:rand"]
-rocksdb = ["dep:rocksdb", "storage"]
+rocksdb = ["storage", "dep:rocksdb"]
 serde = [
-    "serde_repr",
     "hashbrown/serde",
     "packable/serde",
     "primitive-types/serde_no_std",
     "iota-crypto/serde",
     "zeroize?/serde",
```
Comment (thibault-martinez) [inline sdk/Cargo.toml:152]: Just wondering, can you do `"dep:zeroize?/serde"`?

File: sdk/src/utils/mod.rs (hunk 1 above)
Comment (DaughterOfMars) [reply sdk/src/utils/mod.rs]: good idea

File: sdk/Cargo.toml (hunk 2 above)
Comment (DaughterOfMars) [reply sdk/Cargo.toml:152]: I don't think so

**Review by thibault-martinez — APPROVED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Failing CI

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**